Tutorial: How to do time-based breakdowns (hour, minute, real time) (#6393)
PostHog/posthog.com: contents/tutorials/time-breakdowns.md

---
title: How to do time-based breakdowns (hour, minute, real time)
date: 2023-07-27
author: ["ian-vanagas"]
showTitle: true
sidebar: Docs
featuredImage: ../images/tutorials/banners/tutorial-6.png
tags: ['hogql', 'insights']
---

By default, PostHog provides an easy way to group events by week, day, and even hour. Sometimes, smaller, more specific breakdowns are required. With [HogQL](/docs/hogql), you can break down events by time of day, hourly, and even minute-by-minute to help you do a detailed analysis of when they happen, and this tutorial shows you how to do that.

## Time of day breakdown

> We start with an easy scenario, you want to know what part of the day events take place in. To do this, [create an insight](https://app.posthog.com/insights/new), then break down by checking if the event timestamp is in the morning, afternoon, evening, or night. We create a HogQL `multiIf()` expression using the hours of the day to check against `toHour(timestamp)` like this:

Reviewer: ```suggestion
We start with an easy scenario: Breaking down what part of the day events take place in. To do this, we [create an insight](https://app.posthog.com/insights/new) and then break it down by checking if the event timestamp is in the morning, afternoon, evening, or night. We do this by creating a HogQL `multiIf()` expression using the hours of the day to check against `toHour(timestamp)` like so:
```

```sql
multiIf(
   5 >= toHour(timestamp) and toHour(timestamp) < 12, 'morning',
   12 >= toHour(timestamp) and toHour(timestamp) < 17, 'afternoon',
   17 >= toHour(timestamp) and toHour(timestamp) < 20, 'evening',
   'night'
)
```

You can visualize this with a total value bar chart to the sums for the date range or time series bar to see how they change day to day.

![Time of day](../images/tutorials/time-breakdowns/day.png)

> **Note:** Event timestamps default to UTC. You need to adjust your times depending on your local time. For example, if you are in Pacific Daylight Time, subtract 7 hours from UTC timestamps to match your local time.

## Hourly breakdown

> PostHog enables you to group data by hour for single-day date ranges. If you want to sum these values across ranges longer than one day, you can breakdown by `toHour(timestamp)` and, again, use a total value or time series bar chart to visualize.

Reviewer: ```suggestion
PostHog enables you to group data by hour for single-day date ranges. If you want to sum these values across ranges longer than one day, you can break it down by `toHour(timestamp)` and, once again, use a total value or time series bar chart to visualize.
```
